### PR TITLE
[EMB-537] osf-link

### PR DESCRIPTION
**Pull request by brianjgeiger**
## Purpose

To ensure we are using one type of Link in the main app, and to track the clicks with those links and buttons.

## Summary of Changes
* Rename `Link` component to `OsfLink` and standardize the interface just a smidge
* Replace nearly all uses of other linking options in main app to `OsfLink`
* Add `data-analytics-context` identifiers to all pages and some components
* Add `data-analytics-name` identifiers to all clickable items we control in main app
* Use new `click` test helper that verifies `data-analytics-name` existence
* Remove old analytics tracking
* Update handbook and tests

## QA Notes

Links that go to places not within Ember-osf-web are untestable, although there are fewer ways to mess that up than there were before. You can see the minor visual differences at the Percy section of the Checks below, or go to 
https://percy.io/Center-for-Open-Science/ember-osf-web/builds/1325054

There's a list of thing below that aren't well covered by tests, so hypothetically there could be a problem. I'm reasonably certain that testing one will test all of them.

## Ticket

https://openscience.atlassian.net/browse/EMB-537

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

# Not well covered by tests (hrefs)
## Dashboard

- data-analytics-name='noteworthy_search'
- data-analytics-name='meetings_button'
- data-analytics-name='preprints_button'
- Project links in user’s projects
- project links in New and Popular lists
- Institutions
- Create project - Go to new

## Home

- Youtube video
- Explore activity
- Cos.io
- #signup
- Institutions
- institutions read more
- Visit institution
- Contact us

## Sign up page

- ORCid
- Institution

## Settings side nav bar

- addons
- notifications

## Support
- All the things

## Contributors list

- Click on a contributor
- And x more

## File share button

- Twitter
- facebook
- linkedin
- email

## Join OSF Banner

- sign-up
- learn more

## Node card (forks page) dropdown menu

- Contributors
- Settings

## Node and registrations analytics pages navbar

- Settings

## Footer
- All the things

## Nav bar user dropdown

- Sign up
- My profile
- Settings

## Nav bar

- My projects
- Preprints

### CODE REVIEW

File: app/dashboard/template.hbs [hunk 1]
```diff
@@ -1,309 +1,307 @@
...
-    <div class='container p-t-lg'>
-        <div class='row m-t-lg {{if (or (gt this.nodes.length 9) this.loading) (local-class 'quick-search-contents')}}'>
-            <div class='col-md-10 col-md-offset-1 col-lg-8 col-lg-offset-2'>
-                <div class='row'>
-                    <div class='col-xs-12'>
-                        <h2 class='col-xs-9'>{{t 'dashboard.title'}}</h2>
-                        <div class='m-b-lg col-xs-3'>
-                            <div class='pull-right'>
-                                <OsfButton 
-                                    data-test-create-project-modal-button
```
Comment (brianjgeiger) [inline app/dashboard/template.hbs]: Put this back

File: app/dashboard/template.hbs [hunk 2]
```diff
@@ -1,309 +1,307 @@
...
                                                 {{/if}}
-                                            {{/each}}
-                                        </div>
-                                        {{#if this.hasMore}}
-                                            {{#if this.loadingMore}}
-                                                <LoadingIndicator />
-                                            {{else}}
-                                                <div class='text-center'>
-                                                    <OsfButton 
-                                                        data-test-load-more
```
Comment (brianjgeiger) [inline app/dashboard/template.hbs]: Put this back

File: app/register/template.hbs [hunk 3]
```diff
@@ -37,32 +37,30 @@
...
                 >
                     <img
                         src='/assets/images/register/orcid-logo.png'
                         alt={{t 'routes.register.orcid'}}
                         local-class='sign-up-button__logo'
                     >
                     <div local-class='sign-up-button__label'>{{t 'routes.register.orcid'}}</div>
-                </a>
-                <a
-                    data-test-institution-button
```
Comment (brianjgeiger) [inline app/register/template.hbs]: Put this back

File: app/settings/developer-apps/-components/client-secret/template.hbs [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
+            {{fa-icon 'eye-slash'}}
+            {{t 'settings.developer-apps.hideSecret'}}
+        {{else}}
+            {{fa-icon 'eye'}}
+            {{t 'settings.developer-apps.showSecret'}}
+        {{/if}}
+    </OsfButton>
 
-<DeleteButton
-    data-test-reset-client-secret
```
Comment (brianjgeiger) [inline app/settings/developer-apps/-components/client-secret/template.hbs]: Put this back

File: app/settings/developer-apps/create/template.hbs [hunk 5]
```diff
@@ -1,20 +1,18 @@
-{{! template-lint-disable no-implicit-this }}
-<div>
-    {{#link-to data-test-back-to-developer-apps
-        'settings.developer-apps'
-        click=(action 'click' 'link' 'Settings - Developer apps - Create - Back to list' target=this.analytics)
-    }}
+<div data-analytics-scope='Create'>
+    <OsfLink
+        data-analytics-name='Back to list'
```
Comment (brianjgeiger) [inline app/settings/developer-apps/create/template.hbs]: Add back `data-test-back-to-developer-apps`

File: app/settings/developer-apps/edit/template.hbs [hunk 6]
```diff
@@ -1,81 +1,80 @@
-{{! template-lint-disable no-implicit-this }}
-<div>
-    {{#link-to data-test-back-to-developer-apps
-        'settings.developer-apps'
-        click=(action 'click' 'link' 'Settings - Developer apps - Edit - Back to list' target=this.analytics)
-    }}
+<div data-analytics-scope='Edit'>
+    <OsfLink
+        data-analytics-scope='Back to list'
```
Comment (brianjgeiger) [inline app/settings/developer-apps/edit/template.hbs]: Add back `data-test-back-to-developer-apps`

File: app/settings/tokens/create/template.hbs [hunk 7]
```diff
@@ -1,28 +1,26 @@
...
-        {{fa-icon 'arrow-left'}}
-        {{t 'settings.tokens.backToList'}}
-    {{/link-to}}
-</div>
-{{! template-lint-enable no-implicit-this }}
-<h4>{{t 'settings.tokens.createToken'}}</h4>
+<div data-analytics-scope='Create'>
+    <div>
+        <OsfLink
+            data-analytics-name='Back to list'
```
Comment (brianjgeiger) [inline app/settings/tokens/create/template.hbs]: Add back `data-test-back-to-tokens`

File: app/settings/tokens/create/template.hbs [hunk 8]
```diff
@@ -1,28 +1,26 @@
...
-        {{t 'settings.tokens.createToken'}}
-    </OsfButton>
-{{/settings/tokens/-components/token-form}}
\ No newline at end of file
+    {{#settings/tokens/-components/token-form
+        onSave=(action 'onSave')
+        as |form|
+    }}
+        <OsfButton
+            data-analytics-name='Submit button'
```
Comment (brianjgeiger) [inline app/settings/tokens/create/template.hbs]: Add back `data-test-create-token-button`

File: app/settings/tokens/edit/template.hbs [hunk 9]
```diff
@@ -1,66 +1,63 @@
...
-            <DeleteButton
-                @delete={{action this.deleteToken}}
-                @disabled={{form.disabled}}
-                @analyticsScope='Settings - Tokens - Edit'
-                @modalTitle={{t 'settings.tokens.confirmDelete.title' tokenName=this.token.name}}
-                @modalBody={{t 'settings.tokens.confirmDelete.body'}}
+<div data-analytics-scope='Edit'>
+    <div>
+        <OsfLink
+            data-analytics-name='Back to list'
```
Comment (brianjgeiger) [inline app/settings/tokens/edit/template.hbs]: Add back `data-test-back-to-tokens`

File: app/settings/tokens/edit/template.hbs [hunk 10]
```diff
@@ -1,66 +1,63 @@
...
+            }}
+                <DeleteButton
+                    @delete={{action this.deleteToken}}
+                    @disabled={{form.disabled}}
+                    @modalTitle={{t 'settings.tokens.confirmDelete.title' tokenName=this.token.name}}
+                    @modalBody={{t 'settings.tokens.confirmDelete.body'}}
+                />
 
-            <OsfButton
-                data-test-save-token-button
```
Comment (brianjgeiger) [inline app/settings/tokens/edit/template.hbs]: Put this back

File: lib/osf-components/addon/components/file-share-button/component.ts [hunk 11]
```diff
@@ -28,11 +28,6 @@ export default class FileShareButton extends Component {
     styleNamespace?: string;
     elementId!: string;
 
-    // @computed('styleNamespace')
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/file-share-button/component.ts]: Remove this?

File: lib/osf-components/addon/components/new-project-navigation-modal/template.hbs [hunk 12]
```diff
@@ -12,20 +15,20 @@
...
+            data-analytics-name='stay_here'
             @onClick={{action this.stayHere reload=true}}
         >
             {{t 'new_project.stay_here'}}
         </OsfButton>
-        <a
-            data-test-go-to-new
-            href='{{@node.links.html}}'
+        <OsfLink
+            data-analytics-name='go_to_new_project'
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/new-project-navigation-modal/template.hbs]: Add back `data-test-go-to-new`

File: lib/osf-components/addon/components/simple-paginator/template.hbs [hunk 13]
```diff
@@ -1,7 +1,7 @@
 {{#if this.hasMultiplePages}}
     <span local-class='SimplePaginator__element'>
         <OsfButton
-            data-test-previous-page-button
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/simple-paginator/template.hbs]: Put this back

File: lib/osf-components/addon/components/simple-paginator/template.hbs [hunk 14]
```diff
@@ -17,7 +17,7 @@
     </span>
     <span local-class='SimplePaginator__element'>
         <OsfButton
-            data-test-next-page-button
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/simple-paginator/template.hbs]: Put this back

File: app/settings/developer-apps/index/template.hbs [hunk 15]
```diff
@@ -1,27 +1,27 @@
...
-    {{#link-to
-        data-test-create-app-link
-        'settings.developer-apps.create'
-        class='btn btn-primary'
-    }}
-        {{t 'settings.developer-apps.createApp'}}
-    {{/link-to}}
-    {{! template-lint-enable no-implicit-this }}
-</div>
+<div data-analytics-context='Developer Apps'>
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply app/settings/developer-apps/index/template.hbs]: I caught all those the first two times, but apparently I keep doing it. Siigh.

File: tests/integration/helpers/fake-node.ts [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+    belongsTo(relationship: string) {
+        if (relationship === 'parent') {
+            return { id: () => this.parentId };
+        } else if (relationship === 'root') {
+            return { id: () => this.parentId };
+        } else {
+            throw new Error('Unknown relationship passed to FakeNode helper\'s belongsTo().');
+        }
+    }
+}
```
Comment (jamescdavis) [inline tests/integration/helpers/fake-node.ts]: please remove

File: app/settings/tokens/template.hbs [hunk 17]
```diff
@@ -1,7 +1,10 @@
 {{title (t 'settings.tokens.title')}}
 <Panel as |panel|>
     <panel.heading @title={{t 'settings.tokens.title'}} />
-    <panel.body>
+    <panel.body
+        data-analytics-scope='Tokens'
+        class='panel-body'
```
[the remark this answers is not in the document]
Comment (brianjgeiger) [reply app/settings/tokens/template.hbs]: Another merge conflict issue

**Pull request merged**